### PR TITLE
Tests improvements

### PR DESCRIPTION
**Pull request by yodada**
In this PR:
 - import random and set random.seed(42) in *almost* all tests. (https://github.com/pytorch/pytorch/issues/7068)
 - add hypothesis tests to sum/mean/abs/logsoftmax/logsoftmax_backward
 - found a failure in logsoftmax #61

### CODE REVIEW

**Review by vb000 — APPROVED** [review]
Looks good! Thanks for adding the softmax corner case... on it right away.
